content/error-handling/index.textile

```diff
@@ -0,0 +1,130 @@
+
+* Set up a custom log handler to capture and manage errors in a way that suits your requirements.
+* Meta channels allow you to monitor errors that might not otherwise be visible to clients, providing additional insights into issues.
+* The Dev console in your Ably "dashboard":https://ably.com/accounts/any is a quick and easy way to inspect errors and events, especially during development or debugging.
+
+Common errors in Ably-supported apps usually arise from:
+
+- *Authentication issues* := Invalid or expired tokens, incorrect API keys.
+- *Permission errors* := Missing or insufficient capabilities for publishing or subscribing.
+- *Channel attachment failures* := Incorrect channel names, network connectivity problems, or unauthorized access.
+
+h2.(#format) Error format
+
+All errors returned by Ably are standardized and use the @ErrorInfo@ object:
+
+```[text]
+{
+  code: 40005,
+  statusCode: 400,
+  cause: Authentication,
+  nonfatal: false,
+  href: 'https://help.ably.io/error/40005'
+}
+```
+
+The following explains each property of an @ErrorInfo@ object:
+
+- *code* := Ably-specific numeric code indicating the error type.
+- *statusCode* := An HTTP status code providing broader context, such as 400 for a bad request.
+- *cause* := A brief description of the issue.
+- *nonfatal* := A boolean indicating whether the error is critical.
+- *Href* := A direct link to Ably's documentation or for quick troubleshooting references.
+
+h2(#logging). Logging
+
+Most Ably SDKs allow you to customize the function that handles logging. This function is usually set in the options when configuring a client, such as the @ClientOptions@ object for Pub/Sub.
+
+Two separate properties can be set:
+
+- *logHandler* :=  Provide a custom function for each line of log output.
+- *logLevel* := The verbosity of the logging output, from silent to trace. In some SDKs this is numeric, and in others string.
+
+The following example configures the Ably client to log only error messages (level 1) and processes them with the function @logWriteFun()@:
+
+```[text]
+const ablyClient = new AblyClient(ably, {
+  logHandler: logWriteFunc,
+  logLevel: 1 // Errors only
+});
+```
+
+The following is an example log entry:
+
+```[text]
+LOG [Level 1]: Ably: ConnectionManager.failQueuedMessages(): failing 1 queued messages, err = [_ErrorInfo: Application Ptz0jg disabled.
+(See https://help.ably.io/error/40300 for help.); statusCode=403; code=40300]
+```
+
+h2(#meta). [meta]log channels
+
+Errors are not published in the meta channels when the client can be directly notified. For example, if a client publishes a message but exceeds a channel rate limit, the @publish()@ method uses an error callback to notify the client.
```

Comment:
I think you need an intro to what metachannels are before getting into this info. Should we also link out to the metachannels page?